test/adapter/json_api/linked_test.rb

```diff
@@ -277,6 +277,79 @@ def test_nil_link_with_specified_serializer
             assert_equal expected, adapter.serializable_hash
           end
         end
+
+        class NoDuplicatesTest < Minitest::Test
+          Post = Class.new(::Model)
+          Author = Class.new(::Model)
+
+          class PostSerializer < ActiveModel::Serializer
+            type 'posts'
+            class AuthorSerializer < ActiveModel::Serializer
```

Comment:
We need #1225 for that. The reason is that if we do not use nested serializers here, we have a namespacing issue. With #1225, we can declare the serializers in a non-nested way and it will still use the right one.